platform/common/driver/vault.go

```diff
@@ -26,11 +27,17 @@ type VersionedRead struct {
 	TxNum TxNum
```

Comment:
Maybe we should change also `VersionedRead` and `VersionedValue` to contain `Version` instead of block and txnum?